src/common/snippets/include/snippets/pass/mha_tokenization.hpp

```diff
@@ -45,7 +45,50 @@ namespace ov::snippets::pass {
 class TokenizeMHASnippets : public ov::pass::MatcherPass {
 public:
     OPENVINO_MATCHER_PASS_RTTI("snippets::pass::TokenizeMHASnippets");
-    explicit TokenizeMHASnippets(const SnippetsTokenization::Config& config);
+
+    /**
+     * @interface Config
+     * @brief Configuration for TokenizeMHASnippets pass
+     * @ingroup snippets
+     */
+    struct Config : public TokenizationConfig {
+        Config(const TokenizationConfig& tokenization_config,
+               bool enable_transpose_on_output,
+               bool dyn_mha_token,
+               std::set<size_t> mha_transpose_ranks)
+            : TokenizationConfig(tokenization_config),
+              m_mha_token_enable_transpose_on_output(enable_transpose_on_output),
+              m_is_dynamic_mha_token_enabled(dyn_mha_token),
+              m_mha_supported_transpose_ranks(std::move(mha_transpose_ranks)) {}
+
+        [[nodiscard]] bool get_mha_token_enable_transpose_on_output() const {
+            return m_mha_token_enable_transpose_on_output;
+        }
+
+        [[nodiscard]] bool is_dynamic_mha_token_enabled() const {
+            return m_is_dynamic_mha_token_enabled;
+        }
+
+        [[nodiscard]] std::set<size_t> get_mha_supported_transpose_ranks() const {
+            return m_mha_supported_transpose_ranks;
+        }
+
+    private:
+        // False if Transpose on output isn't tokenized in MHA Tokenization.
+        // Otherwise, it may be fused into Subgraph if possible
+        // TODO [111813]: Remove please when the ticket 111813 is implemented
+        bool m_mha_token_enable_transpose_on_output = true;
+        // If True, MHA pattern with dynamic nodes will be tokenized
+        // Otherwise dynamic MHA won't be tokenized
+        // Currently, the flag can be set to `True` only for testing purposes.
```

Comment:
Shall we remove this comment? It seems not relevant anymore. If snippets cache is 0, it is already disabled on plugin side